### PR TITLE
DM-13246: Synchronize DMTN-056 with Butler prototype

### PR DESCRIPTION
**Pull request by pschella**

### CODE REVIEW

File: python/lsst/butler/schema.py [hunk 1]
```diff
@@ -180,8 +182,10 @@
 
 VisitRangeTable = \
     Table('VisitRange', metadata,
-          Column('visit_begin', Integer, primary_key=True, nullable=False),
-          Column('visit_end', Integer, primary_key=True, nullable=False),
+          Column('visit_begin', Integer, ForeignKey(
+              'Visit.visit_number'), primary_key=True, nullable=False),
+          Column('visit_end', Integer, ForeignKey(
+              'Visit.visit_number'), primary_key=True, nullable=False),
```
Comment (TallJimbo) [inline python/lsst/butler/schema.py]: I'm surprised it makes sense to actually make this a foreign key, given that it's defining a range.

**Review by TallJimbo — APPROVED** [review]

File: python/lsst/butler/schema.py (hunk 1 above)
Comment (pschella) [reply python/lsst/butler/schema.py]: I was assuming the range had to be valid. So both need to exist. But perhaps this is not true?

Comment (TallJimbo) [reply python/lsst/butler/schema.py]: I think we'll want at least half-open ranges, presumably marked by null endpoints.

Comment (pschella) [reply python/lsst/butler/schema.py]: Ok, we'll remove the foreign key checks and hope for the best.

**Pull request merged**